tests/cases/compiler/noImplicitReturnsExclusions.ts

```diff
@@ -58,3 +58,39 @@ function f16(b: boolean) {  // Error
     if (b) return 42;
     if (b) return;
 }
+
+declare class HistoryItem {
+  input: {
+    location: {
+      uri: string;
+    };
+  };
+}
+
+interface Thenable<T> {
+  then<TResult>(
+    onfulfilled?: (value: T) => TResult | Thenable<TResult>,
+    onrejected?: (reason: any) => TResult | Thenable<TResult>
+  ): Thenable<TResult>;
+  then<TResult>(
+    onfulfilled?: (value: T) => TResult | Thenable<TResult>,
+    onrejected?: (reason: any) => void
+  ): Thenable<TResult>;
+}
+
+export declare function executeCommand<T = unknown>(
+  command: string,
+  ...rest: any[]
+): Thenable<T>;
+
+export declare function registerCommand(
+  command: string,
+  callback: (...args: any[]) => any,
+  thisArg?: any
+): void;
+
+registerCommand("_references-view.showHistoryItem", async (item) => { // Error, contextual return type of Promise<unknown>

```

Comment:
When I created this PR, this code wasn't erroring. My initial changes created a regression with this code and it started to error (problem caught by the extended test suite). However, right now it is **supposed** to error as per the change [here](https://github.com/microsoft/TypeScript/pull/53607/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R35668-R35670). I decided to add this as an extra regression test.